src/main/java/FileSieve/BusinessLogic/FileManagement/ConcurrentFileManager.java

```
@@ -0,0 +1,389 @@
+                    ) {
+                        while ((sourceByte = bis.read()) != -1) {
+                            bos.write(sourceByte);
+
+                            totalPercentCopied = (int) (++copiedBytes * 100 / totalBytes);
+                            if ((getProgress() != totalPercentCopied) && (totalPercentCopied != 100)) {
+                                setProgress(totalPercentCopied);
+                                publish(new SimpleImmutableEntry(this.targetPathname, getProgress()));
+                            }
+
+                            filePercentCopied = (int) (++soFar * 100 / fileBytes);
+                            if ((fileProgress != filePercentCopied) && (filePercentCopied != 100)) {
+                                fileProgress = filePercentCopied;
+                                publish(new SimpleImmutableEntry(targetPathname, fileProgress));
+                            }
+                        }
+
+                        fileProgress = 100;
+                        publish(new SimpleImmutableEntry(targetPathname, 100));
+
+                    } catch (SecurityException | IOException e) {
+                        noErrors = false;
+
+                        setProgress((int) ((totalPercentPreviouslyCopied + fileBytes) * 100 / totalBytes));
+                        publish(new SimpleImmutableEntry(this.targetPathname, getProgress()));
+                        publish(new SimpleImmutableEntry(targetPathname, null));
+
+                        try {
+                            if (Files.exists(targetPathname) && ((targetPathname.toFile().length() == 0L) || (soFar > 0))) {
+                                Files.delete(targetPathname);
+                            }
+                        } catch (SecurityException | IOException ex) {
+                            // Ignore exceptions - this is a best attempt at deleting the written file, which was cut short
```

Never, ever ignore exceptions. 
Exceptions must be handled in the code or passed up to the client caller. 
Ignoring a problem is not a valid way to  handle it. 
Ignoring an exception is called 'dropping an exception' 
Exceptions are like eggs. You want to handle them carefully, never drop them, and wash you hands after using them. - Wait, not that last part, but the rest is really true. 
Otherwise, you could make yourself sick spending hours debugging a problem because the issue was carefully ignored somewhere deep in the code. This has happened to be. Well, not the sick part, unless you count sick with rage at the person who 'dropped the exception' and made debugging so much harder. 
Note: sometimes it does make sense to convert the checked exception to a runtime exception. The times this makes sense is when there's no possible way client code can deal with the problem effectively. Note sometimes != always or most of the time. More like once in a while. This might be one of those cases though. Not sure.